### PR TITLE
send final result when stream is true

### PR DESCRIPTION
**Pull request by likev**
send final result(which contain important info such as conversationId) when stream is true

### CODE REVIEW

File: bin/server.js [hunk 1]
```diff
@@ -108,7 +108,9 @@ server.post('/conversation', async (request, reply) => {
 
     if (result !== undefined) {
         if (body.stream === true) {
+            reply.sse({event:'RESULT', id: '', data: JSON.stringify(result) });
```
Comment (waylaidwanderer) [inline bin/server.js:111]:
```suggestion
            reply.sse({ event: 'result', id: '', data: JSON.stringify(result) });
```

**Comment by waylaidwanderer** [comment]
@likev could you also update the examples and readme too?

**Comment by likev** [comment]
> uld you also update the examples and readme too?

@waylaidwanderer sure, i'll try my best

**Comment by likev** [comment]
@waylaidwanderer Example and README.md of stream API are updated.

File: README.md [hunk 2]
```diff
@@ -290,6 +290,8 @@ Alternatively, you can install and run the package directly.
     - using `docker-compose up` (requires Docker)
 
 #### Usage
+<details>
+<summary><strong>Method1</strong></summary>
```
Comment (waylaidwanderer) [inline README.md:294]:
```suggestion
<summary><strong>Method 1</strong></summary>

```

File: README.md [hunk 3]
```diff
@@ -332,15 +334,26 @@ If there was an error sending the message to ChatGPT:
     "error": "There was an error communicating with ChatGPT."
 }
 ```
-
+</details>
+<details>
+<summary><strong>Method2</strong></summary>
```
Comment (waylaidwanderer) [inline README.md:339]:
```suggestion
<summary><strong>Method 2</strong></summary>

```

File: README.md [hunk 4]
```diff
@@ -357,6 +370,7 @@ Successful output:
 { data: ' you', event: '', id: '', retry: undefined }
 { data: ' today', event: '', id: '', retry: undefined }
 { data: '?', event: '', id: '', retry: undefined }
+{ data: 'Object JSON', event: 'result', id: '', retry: undefined }
```
Comment (waylaidwanderer) [inline README.md:373]:
```suggestion
{ data: '<result JSON here>', event: 'result', id: '', retry: undefined }
```

File: README.md [hunk 5]
```diff
@@ -374,6 +388,11 @@ if (message.event === 'error') {
   console.error(JSON.parse(message.data).error); // There was an error communicating with ChatGPT.
 }
 ```
+</details>
+
+#### Notes
+- Method1 is simple but Time to First Byte (TTFB) is long.
```
Comment (waylaidwanderer) [inline README.md:394]:
```suggestion
- Method 1 is simple, but Time to First Byte (TTFB) is long.
```

File: README.md [hunk 6]
```diff
@@ -374,6 +388,11 @@ if (message.event === 'error') {
...
+- Method2 use non-standard implementation of [server-sent event API](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events), you should import `fetch-event-source` first and use `POST` method.
```
Comment (waylaidwanderer) [inline README.md:395]:
```suggestion
- Method 2 uses a non-standard implementation of [server-sent event API](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events); you should import `fetch-event-source` first and use `POST` method.
```

**Review by waylaidwanderer — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -290,6 +290,9 @@ Alternatively, you can install and run the package directly.
     - using `docker-compose up` (requires Docker)
 
 #### Usage
+<details>
+<summary><strong>Method 1</strong></summary>
```
Comment (waylaidwanderer) [inline README.md:294]:
```suggestion
<summary><strong>Method 1 (POST)</strong></summary>
```

File: README.md [hunk 8]
```diff
@@ -332,15 +335,27 @@ If there was an error sending the message to ChatGPT:
     "error": "There was an error communicating with ChatGPT."
 }
 ```
+</details>
+<details>
+<summary><strong>Method 2</strong></summary>
```
Comment (waylaidwanderer) [inline README.md:340]:
```suggestion
<summary><strong>Method 2 (SSE)</strong></summary>
```

**Review by waylaidwanderer — CHANGES_REQUESTED** [review]

**Review by waylaidwanderer — APPROVED** [review]

**Comment by waylaidwanderer** [comment]
@likev This is great, thanks so much.

**Pull request merged**